### PR TITLE
Simplified some code

### PR DESCRIPTION
**Pull request by tanelihuuskonen**
Simplified a few bits of code in BitArray and RGBLuminanceSource, improving performance at least marginally.

### CODE REVIEW

File: core/src/main/java/com/google/zxing/common/BitArray.java [hunk 1]
```diff
@@ -163,15 +163,8 @@ public void setRange(int start, int end) {
...
-      if (firstBit == 0 && lastBit == 31) {
-        mask = -1;
-      } else {
-        mask = 0;
-        for (int j = firstBit; j <= lastBit; j++) {
-          mask |= 1 << j;
-        }
-      }
+      // Ones from firstBit to lastBit, inclusive
+      int mask = (2 << lastBit) - (1 << firstBit);
```
Comment (srowen) [inline core/src/main/java/com/google/zxing/common/BitArray.java]: That's pretty cool, never though of writing it this way. Seems to work. You can inline this in the line below if you like.

**Comment by srowen** [comment]
Yeah looks like nice optimizations, well done. I'll merge it when I get back.

File: core/src/main/java/com/google/zxing/common/BitArray.java (hunk 1 above)
Comment (tanelihuuskonen) [inline core/src/main/java/com/google/zxing/common/BitArray.java]: Right, that makes sense.  Done.

**Comment by tanelihuuskonen** [comment]
Thanks.  I like that kind of stuff.

**Pull request merged**